scipy/linalg/_basic.py

```diff
@@ -243,6 +276,20 @@ def solve(a, b, lower=False, overwrite_a=False,
                                  overwrite_b=overwrite_b)
         _solve_check(n, info)
         rcond, info = sycon(lu, ipvt, anorm)
+    # Diagonal case
+    elif assume_a == 'dia':
+        x = (b1.T / np.diag(a1)).T
+    # Tri-diagonal case
+    elif assume_a == 'tri':
+        a1 = a1.T if transposed else a1
+        dl, d, du = np.diag(a1, -1), np.diag(a1, 0), np.diag(a1, 1)
+        _gtsv = get_lapack_funcs('gtsv', (a1, b1))
+        x, info = _gtsv(dl, d, du, b1, False, False, False, overwrite_b)[3:]
```

Comment:
is there a `??con` routine for tri-diagonal? I couldn't find it.

Reply:
No but there is an O(n) algo given in https://doi.org/10.1137/S0895479897314747 I should have it somewhere in my old computer. I'll check.

Reply:
Oh, there is: [`?GTCON`](https://www.netlib.org/lapack/explore-html/da/d76/group__gtcon_ga005ee08a820a0e5a6022b4a3d55d50f0.html#ga005ee08a820a0e5a6022b4a3d55d50f0).